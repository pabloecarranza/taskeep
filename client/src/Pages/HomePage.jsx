import React from 'react';
import { Sidebar } from './../components/Sidebar';
import { TaskArea } from './../components/TaskArea';
import { Box, Flex } from '@chakra-ui/react';

export const HomePage = () => {
	return (
		<Flex justifyContent='center' mt='5px'>
			<Sidebar />
			<TaskArea />
		</Flex>
	);
};
