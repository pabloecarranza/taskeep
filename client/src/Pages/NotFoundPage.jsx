import {
	Heading,
	Container,
	Box,
	Center,
	Button,
	Text,
	Image,
} from '@chakra-ui/react';
import Logo from '../assets/daily-tasks.png';
import { ArrowForwardIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

export const NotFoundPage = ({
	error_text,
	message_redirect,
	page_redirect,
	test = false,
}) => {
	let navigate = useNavigate();
	const reDirectTo = () => {
		navigate('/homepage', { replace: true });
	};

	return (
		<Container centerContent>
			<Center padding='4' w='100vw' h='100vh'>
				<Box
					w='80%'
					h='80%'
					display='flex'
					flexDir='column'
					alignItems='center'
					justifyContent='center'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Heading as='h2' size='2xl' fontWeight='bold' textAlign='center'>
						{error_text}
						<Text
							bgGradient={test ? '' : 'linear(to-l, #7928ca, #0084ff)'}
							bgClip='text'
							fontSize='3xl'
							fontWeight='bold'
							pb='30px'
							pt='20px'
						>
							{message_redirect}
						</Text>
					</Heading>
					<Box>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							size='lg'
							onClick={reDirectTo}
						>
							{page_redirect}
						</Button>
					</Box>
				</Box>
			</Center>
		</Container>
	);
};

NotFoundPage.propTypes = {
	error_text: PropTypes.string.isRequired,
	message_redirect: PropTypes.string.isRequired,
	page_redirect: PropTypes.string.isRequired,
};
