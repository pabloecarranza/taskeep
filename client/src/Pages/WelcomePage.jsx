import {
	Box,
	Flex,
	Text,
	Heading,
	Button,
	Center,
	Image,
	useDisclosure,
	ScaleFade,
	Link,
} from '@chakra-ui/react';
import { ExternalLinkIcon } from '@chakra-ui/icons';
import { BsLinkedin, VscGithubInverted } from 'react-icons/all';
import Logo from '../assets/daily-tasks.png';
import Typewriter from 'typewriter-effect';
import { LoginForm } from '../components/LoginForm';
import { ArrowForwardIcon } from '@chakra-ui/icons';

import bg from '../assets/hojas.png';

export const WelcomePage = () => {
	const { isOpen, onToggle } = useDisclosure();

	return (
		<Flex
			w='100vw'
			h='100vh'
			flexDir={{ base: 'column', md: 'row', lg: 'row' }}
		>
			<Box w='100%' h='100%'>
				<Center
					w='100%'
					h='100%'
					flexDir='column'
					alignItems='flex-end'
					textAlign='right'
					pr='35px'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Text
						bgGradient='linear(to-l, #b785e9, #61acf3)'
						bgClip='text'
						fontSize='6xl'
						fontWeight='extrabold'
						pb='10px'
					>
						TASKEEP
					</Text>
					<Heading as='h2' size='2xl' noOflines={1} fontWeight='bold'>
						<Typewriter
							options={{
								strings: ['The most powerful', 'The real powerful'],
								autoStart: true,
								loop: true,
							}}
						/>

						<Text
							bgGradient='linear(to-l, #7928ca, #0084ff)'
							bgClip='text'
							fontSize='5xl'
							fontWeight='bold'
							pb='50px'
						>
							Task Management App
						</Text>
					</Heading>
					<Box>
						<Button
							size='lg'
							colorScheme='blue'
							rightIcon={<VscGithubInverted />}
							mr='10px'
							onClick={() =>
								window.open(
									'https://github.com/pabloecarranza/taskeep',
									'_blank'
								)
							}
						>
							GitHub
						</Button>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							variant='outline'
							size='lg'
							onClick={onToggle}
						>
							Get Started
						</Button>
					</Box>

					<Button
						colorScheme='gray'
						variant='link'
						rightIcon={<BsLinkedin />}
						pt='35px'
						onClick={() =>
							window.open(
								'https://www.linkedin.com/in/pabloecarranza/',
								'_blank'
							)
						}
					>
						Made by Pablo Carranza
					</Button>
				</Center>
			</Box>
			<Center
				w='80%'
				h='100%'
				backgroundImage={bg}
				backgroundPosition='center'
				backgroundRepeat='no-repeat'
				backgroundSize='cover'
			>
				<ScaleFade initialScale={0.9} in={isOpen}>
					<LoginForm />
				</ScaleFade>
			</Center>
		</Flex>
	);
};
