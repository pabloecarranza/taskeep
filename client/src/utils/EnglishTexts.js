export const NotFoundPageDates = {
	error_text: "We're sorry - something's gone wrong.",
	message_redirect:
		'Our team has been notified, but click below to go HomePage',
	page_redirect: 'HomePage',
};

export const WelcomePagesDates = {
	app_name: 'TASKEEP',
	slogan_one: 'The most powerful',
	slogan_two: 'The real powerful',
	subtitle: 'Task Management App',
	repository: 'Github',
	start_app: 'Get Starter',
	made: 'Made by Pablo Carranza',
	test: false,
};

export const ModalWelcomeDates = {
	greeting: 'Hi',
	welcome_text_1: 'Welcome to my personal app for task management.',
	welcome_text_2:
		'The main functionalities are to create tasks and group them by lists, set the priority and expiration date. I hope you enjoy it !',
	name: 'Pablo',
	button_text: 'Close',
};

export const ModalConfirmDates = {
	title_partOne: 'Delete ',
	title_partTwo: 'task list',
	message:
		"Are you sure? You can't undo this action afterwards and all task assigned to this list be eliminated",
	button_one: 'Cancel',
	button_two: 'Delete',
};

export const ModalAddListDates = {
	title: 'Create a new task list',
	name: 'List name',
	button_one: 'Cancel',
	button_two: 'Save',
	place_holder: 'Write here...',
};
