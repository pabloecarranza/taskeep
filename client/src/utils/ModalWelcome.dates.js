export const ModalWelcomeDates = {
	greeting: 'Hi',
	welcome_text_1: 'Welcome to my personal app for task management.',
	welcome_text_2:
		'The main functionalities are to create tasks and group them by lists, set the priority and expiration date. I hope you enjoy it !',
	name: 'Pablo',
};
