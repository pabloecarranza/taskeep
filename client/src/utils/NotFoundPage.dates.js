export const NotFoundPageDates = {
	error_text: "We're sorry - something's gone wrong.",
	message_redirect:
		'Our team has been notified, but click below to go HomePage',
	page_redirect: 'HomePage',
};
