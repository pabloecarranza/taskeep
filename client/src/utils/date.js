export function fecha() {
	const meses = [
		'January',
		'February',
		'March',
		'April',
		'May',
		'June',
		'July',
		'August',
		'September',
		'Octuber',
		'November',
		'December',
	];

	const dias_semana = [
		'Sunday',
		'Monday',
		'Tuesday',
		'Wednesday',
		'Thursday',
		'Friday',
		'Saturday',
	];

	const fecha = new Date();

	let date =
		dias_semana[fecha.getDay()] +
		', ' +
		fecha.getDate() +
		' de ' +
		meses[fecha.getMonth()] +
		' de ' +
		fecha.getUTCFullYear();

	return date;
}

export function validate(input) {
	let errors = {};
	const regexName = /^([a-zA-Z ]+)$/i;
	input.name
		? (errors.name = '')
		: (errors.name = 'El nombre no puede estar vacio');
	if (input.name && !regexName.test(input.name)) {
		errors.name = 'El nombre no puede incluir caracteres especiales o numeros';
	}

	return errors;
}

export function diaDeSemana() {
	const dias_semana = [
		'Sunday',
		'Monday',
		'Tuesday',
		'Wednesday',
		'Thursday',
		'Friday',
		'Saturday',
	];

	const fecha = new Date();

	let date = dias_semana[fecha.getDay()];

	return date;
}

export function MesyDia() {
	const meses = [
		'January',
		'February',
		'March',
		'April',
		'May',
		'June',
		'July',
		'August',
		'September',
		'Octuber',
		'November',
		'December',
	];

	const fecha = new Date();

	let date = fecha.getDate() + ' de ' + meses[fecha.getMonth()];
	return date;
}
