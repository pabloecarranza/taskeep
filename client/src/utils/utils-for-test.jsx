import React from 'react';
import { render } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { authSlice } from '../features/api/authSlice';
import { listSlice } from '../features/api/listSlice';
import { sessionSlice } from '../features/api/sessionSlice';
import { taskSlice } from '../features/api/taskSlice';
// As a basic setup, import your same slice reducers

export function renderWithProviders(
	ui,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = configureStore({
			reducer: {
				authSlice: authSlice.reducer,
				listSlice: listSlice.reducer,
				taskSlice: taskSlice.reducer,
				session: sessionSlice.reducer,
			},
			preloadedState,
		}),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}

	// Return an object with the store and all of RTL's query functions
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
