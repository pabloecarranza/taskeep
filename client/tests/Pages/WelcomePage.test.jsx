import { render, screen } from '@testing-library/react';
import { vi, expect, describe, it } from 'vitest';
import { WelcomePage } from '../../src/Pages/WelcomePage';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithProviders } from '../../src/utils/utils-for-test';

describe('Test suite on WelcomePage component', () => {
	const WelcomePagesDates = {
		app_name: 'TASKEEP',
		slogan_one: 'The most powerful',
		slogan_two: 'The real powerful',
		subtitle: 'Task Management App',
		repository: 'Github',
		start_app: 'Get Starter',
		made: 'Made by Pablo Carranza',
		test: true,
	};

	vi.mock('../../src/components/LoginForm', () => {
		return {
			__esModule: true,
			default: () => {
				return <div />;
			},
			LoginForm: () => {
				return <div />;
			},
		};
	});

	it('must match with the snapshot', () => {
		const { container } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);
		expect(container).toMatchSnapshot();
	});

	it('must show the name app send by props', () => {
		const { getByText, getByTestId } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);
		expect(getByText(WelcomePagesDates.app_name)).toBeTruthy();

		expect(getByTestId('app_name').innerHTML).toBe(WelcomePagesDates.app_name);
	});

	it('must show text of subtitle send by props', () => {
		const { getByText } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);

		expect(getByText(WelcomePagesDates.subtitle)).toBeTruthy();
	});

	it('must show the typewriter send by props', () => {
		const { getByText, getByTestId } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);
		expect(getByTestId('typewriter-wrapper')).toBeTruthy();
	});

	it('must show the name of repository send by props', () => {
		const { getByText } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);
		expect(getByText(WelcomePagesDates.repository));
	});

	it('must show text get starter send by props', () => {
		const { getByText } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);
		expect(getByText(WelcomePagesDates.start_app));
	});

	it('must show the text (made by) send by props', () => {
		const { getByText } = renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);
		expect(getByText(WelcomePagesDates.made));
	});

	it('should display Get Starter button', () => {
		renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);

		const button = screen.getByText(WelcomePagesDates.start_app);
		expect(button.textContent).toEqual(WelcomePagesDates.start_app);
	});

	it('should display Repository button', () => {
		renderWithProviders(
			<Router>
				<WelcomePage {...WelcomePagesDates} />
			</Router>
		);

		const button = screen.getByText(WelcomePagesDates.repository);
		expect(button.textContent).toEqual(WelcomePagesDates.repository);
	});
});
