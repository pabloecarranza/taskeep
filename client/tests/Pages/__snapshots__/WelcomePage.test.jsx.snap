// Vitest Snapshot v1

exports[`Test suite on WelcomePage component > must match with the snapshot 1`] = `
<div>
  <div
    class="css-52m22g"
  >
    <div
      class="css-uwwqev"
    >
      <div
        class="css-eo0gac"
      >
        <div
          class="css-0"
        >
          <img
            alt="Logo_Task"
            class="chakra-image css-50xotc"
            src="/src/assets/daily-tasks.png"
          />
        </div>
        <p
          class="chakra-text css-1e0y8hh"
          data-testid="app_name"
        >
          TASKEEP
        </p>
        <h2
          class="chakra-heading css-in3yi3"
          nooflines="1"
        >
          <div
            class="Typewriter"
            data-testid="typewriter-wrapper"
          >
            <span
              class="Typewriter__wrapper"
            />
            <span
              class="Typewriter__cursor"
            >
              |
            </span>
          </div>
          <p
            class="chakra-text css-1um6xi5"
          >
            Task Management App
          </p>
        </h2>
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-1uiavz5"
            type="button"
          >
            Github
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.976 0A7.977 7.977 0 0 0 0 7.976c0 3.522 2.3 6.507 5.431 7.584.392.049.538-.196.538-.392v-1.37c-2.201.49-2.69-1.076-2.69-1.076-.343-.93-.881-1.175-.881-1.175-.734-.489.048-.489.048-.489.783.049 1.224.832 1.224.832.734 1.223 1.859.88 2.3.685.048-.538.293-.88.489-1.076-1.762-.196-3.621-.881-3.621-3.964 0-.88.293-1.566.832-2.153-.05-.147-.343-.978.098-2.055 0 0 .685-.196 2.201.832.636-.196 1.322-.245 2.007-.245s1.37.098 2.006.245c1.517-1.027 2.202-.832 2.202-.832.44 1.077.146 1.908.097 2.104a3.16 3.16 0 0 1 .832 2.153c0 3.083-1.86 3.719-3.62 3.915.293.244.538.733.538 1.467v2.202c0 .196.146.44.538.392A7.984 7.984 0 0 0 16 7.976C15.951 3.572 12.38 0 7.976 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            Get Starter
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
        <button
          class="chakra-button css-1g8yexg"
          type="button"
        >
          Made by Pablo Carranza
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 1.146C0 .513.526 0 1.175 0h13.65C15.474 0 16 .513 16 1.146v13.708c0 .633-.526 1.146-1.175 1.146H1.175C.526 16 0 15.487 0 14.854V1.146zm4.943 12.248V6.169H2.542v7.225h2.401zm-1.2-8.212c.837 0 1.358-.554 1.358-1.248-.015-.709-.52-1.248-1.342-1.248-.822 0-1.359.54-1.359 1.248 0 .694.521 1.248 1.327 1.248h.016zm4.908 8.212V9.359c0-.216.016-.432.08-.586.173-.431.568-.878 1.232-.878.869 0 1.216.662 1.216 1.634v3.865h2.401V9.25c0-2.22-1.184-3.252-2.764-3.252-1.274 0-1.845.7-2.165 1.193v.025h-.016a5.54 5.54 0 0 1 .016-.025V6.169h-2.4c.03.678 0 7.225 0 7.225h2.4z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="css-hu7s1b"
    >
      <div
        class="chakra-offset-slide"
        style="opacity: 0; transform: scale(0.9) translateZ(0);"
      >
        <div
          class="css-1feu3yv"
        >
          <div
            style="opacity: 1;"
          >
            <div
              class="css-zarp5x"
            >
              <div
                class="css-0"
              >
                <svg
                  fill="currentColor"
                  height="50px"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="50px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-1n49pl7"
              >
                Sign In
              </p>
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-0"
                >
                  You don't have an account?
                  <button
                    class="chakra-button css-jje6zt"
                    type="button"
                  >
                    Sign Up.
                  </button>
                </p>
              </div>
            </div>
            <div
              class="css-1r4pumt"
            >
              <div
                class="css-1krjhfx"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="field-:r1:"
                    id="field-:r1:-label"
                  >
                    User Name
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <input
                    aria-required="true"
                    class="chakra-input css-0"
                    id="field-:r1:"
                    name="username"
                    placeholder="Name"
                    required=""
                    type="username"
                    value=""
                  />
                </div>
                <div
                  class="chakra-form-control css-542elh"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="field-:r2:"
                    id="field-:r2:-label"
                  >
                    Password
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      aria-required="true"
                      class="chakra-input css-0"
                      id="field-:r2:"
                      name="password"
                      placeholder="Password"
                      required=""
                      type="text"
                      value=""
                    />
                    <div
                      class="chakra-input__right-element css-1x84ua"
                    >
                      <button
                        class="chakra-button css-xsdlqf"
                        type="button"
                      >
                        Hide
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-8pjhjp"
              >
                <button
                  class="chakra-button css-1i9ssx"
                  type="button"
                >
                  Enter as a guest
                </button>
                <button
                  class="chakra-button css-15j421h"
                  type="button"
                >
                  Log In
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
